### PR TITLE
Create filter to record PLAY_SESSION cookie size

### PR DESCRIPTION
**Pull request by yotommy**
### Description

Create filter to record the size of the `PLAY_SESSION` cookie.

There is an upper limit of 4096 bytes, and we'd like to be able to see where we stand.

Tested on a dev instance and can see recorded metrics.

image

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by yotommy** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5572** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5572" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/5572?utm_source=stack-comment).

File: server/app/filters/RecordCookieSizeFilter.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package filters;
+
+import io.prometheus.client.Histogram;
+import play.mvc.EssentialAction;
+import play.mvc.EssentialFilter;
+import play.routing.HandlerDef;
+import play.routing.Router;
+
+public class RecordCookieSizeFilter extends EssentialFilter {
```
Comment (bion) [inline server/app/filters/RecordCookieSizeFilter.java:13]: **suggestion (blocking):** Please add javadoc and make this class final

**Review by bion — CHANGES_REQUESTED** [review]

File: server/app/filters/RecordCookieSizeFilter.java (hunk 1 above)
Comment (yotommy) [reply server/app/filters/RecordCookieSizeFilter.java]: Done

**Review by bion — APPROVED** [review]

**Review by dkatzz — APPROVED** [review]
Just checking this still works with the metrics enabled turned off too? I think it should, but I know in some cases we do a check before setting it checkNotNull(config).getBoolean("civiform_server_metrics_enabled");

**Comment by yotommy** [comment]
Good question: I ran the server with metrics disabled without a problem.

File: server/test/filters/RecordCookieSizeFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    // Run filter with several requests with various cookie sizes.
+    RecordCookieSizeFilter filter = new RecordCookieSizeFilter();
+    List<Integer> cookieSizes = ImmutableList.of(1000, 2000, 2000, 3000, 3500, 4000);
+    for (int cookieSize : cookieSizes) {
+      assertThat(runFilterWithCookieSize(filter, cookieSize).status()).isEqualTo(200);
+    }
+
+    var metricFamilySamples = CollectorRegistry.defaultRegistry.metricFamilySamples();
+    assertThat(metricFamilySamples.hasMoreElements()).isTrue();
+    var mfs = metricFamilySamples.nextElement();
```
Comment (dkatzz) [inline server/test/filters/RecordCookieSizeFilterTest.java:36]: **nitpick:** Thoughts on renaming this to something clearer? Maybe metricElement?

**Review by dkatzz — APPROVED** [review]
Thanks for adding the tests!

File: server/test/filters/RecordCookieSizeFilterTest.java (hunk 2 above)
Comment (yotommy) [reply server/test/filters/RecordCookieSizeFilterTest.java]: I agree this should be clearer.

I had some trouble with the names. On line 34, the type is `Enumeration<MetricFamilySamples>`. So on line 36, the type is `MetricFamilySamples`.

I'll rename the variables to match the types better.

**Pull request merged**